### PR TITLE
Simplify test a bit

### LINKED ISSUE #246: Flakey Test
These 2 drone builds were run on the same code, one failed and one passed: 
https://drone.grafana.net/grafana/opensearch-datasource/465
https://drone.grafana.net/grafana/opensearch-datasource/466

```
Summary of all failing tests
FAIL src/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/index.test.tsx (38.745 s)
  ● Settings Editor › Raw Data › Should correctly render the settings editor and trigger correct state changes

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

       9 | describe('Settings Editor', () => {
      10 |   describe('Raw Data', () => {
    > 11 |     it('Should correctly render the settings editor and trigger correct state changes', async () => {
         |     ^
      12 |       const metricId = '1';
      13 |       const initialSize = '500';
      14 |       const initialOrder = 'desc';

      at src/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/index.test.tsx:11:5
      at src/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/index.test.tsx:10:3
      at Object.<anonymous> (src/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/index.test.tsx:9:1)
```

### PR DESCRIPTION
**Pull request by sarahzinger**
Fixes https://github.com/grafana/opensearch-datasource/issues/246

Hopefully this is a fix for a flakey test we saw. Unfortunately I haven't reproduced the flakiness on my machine yet. I did attempt to simplify this test and break it into 2 which I think might help.

### CODE REVIEW

File: src/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/index.test.tsx [hunk 1]
```diff
@@ -1,113 +1,107 @@
...
       );
 
-      let settingsButtonEl = screen.getByRole('button', {
-        name: /Size: \d+, Order: \w+$/i,
-      });
-
       // The metric row should have a settings button
-      expect(settingsButtonEl).toBeInTheDocument();
-      expect(settingsButtonEl.textContent).toBe(`Size: ${initialSize}, Order: ${initialOrder}`);
+      expect(screen.getByText('Size: 5, Order: desc')).toBeInTheDocument();
```
Comment (kevinwcyu) [inline src/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/index.test.tsx:39]: can you use [`screen.debug()`](https://testing-library.com/docs/dom-testing-library/api-debugging/#screendebug) to print out the component tree to get a sense of how big it is?

also can you time how long the difference is between using `getByRole` vs `getByText`?

**Review by kevinwcyu — COMMENTED** [review]
do you have any links to test runs that were flakey?

like the simplified tests, but not sure about using `getByText` over `getByRole`

**Comment by sarahzinger** [comment]
Converting back to a draft because I'm unclear if this does anything all that useful, sharing my learnings though in case it jogs other thoughts.

Seems like my new tests is not particularly faster than the old test, but I was able to confirm that both get by text and get by test id seem to be much faster than get by role which is kind of interesting

Screenshot 2023-09-20 at 3 32 36 PM

**Comment by sarahzinger** [comment]
going to close this and put the ticket back in the backlog for now. Maybe someone else will be inspired and do something different with it.

**Pull request closed without merging**